### ISSUE TITLE
Invalid chmod call in classes/framework.php

### ISSUE THREAD

**Issue by marsh0lion** [h5p/moodle-mod_hvp#471]
While trying to install a H5P editor, i noticed I was getting the following JavaScript error:
```
TypeError: e.contentTypes is undefined
    value h5p-hub-client.js:15505
    ....
```

I found out this was because `chmod` was being used incorrectly in classes/framework.php line 247 where it was trying to change permissions on a stream object (as supplied by fopen).

**Comment by abautu** [comment]
I confirm this problem and that the fix is working for it (Moodle 4.0.5, PHP 8.0). This patch also seem to fix what is reported in #465.

**Comment by samuelchchan** [comment]
Just want to check when will this fix be pushed out? As I am having exact same issue

**Comment by davewallace** [comment]
I am also experiencing the same issue, any progress?

**Comment by benfes** [comment]
The patch for this ticket seems to have remedied a number of issues that people are reporting. Wondering if this can actually be merged? Without this change the editor has now way of installing any h5p libraries from the hub etc. Only able to upload .h5p files without this change.

**Comment by icc** [comment]
Fixed by #472 will release an updated plugin asap.
